src/test/java/com/fasterxml/jackson/databind/deser/creators/AnySetterForCreator562Test.java

```diff
@@ -0,0 +1,180 @@
+import static org.junit.jupiter.api.Assertions.fail;
+
+// [databind#562] Allow @JsonAnySetter on Creator constructors
+public class AnySetterForCreator562Test extends DatabindTestUtil
+{
+    static class POJO562
+    {
+        String a;
+        Map<String,Object> stuff;
+
+        @JsonCreator
+        public POJO562(@JsonProperty("a") String a,
+            @JsonAnySetter Map<String, Object> leftovers
+        ) {
+            this.a = a;
+            stuff = leftovers;
+        }
+    }
+
+    static class PojoWithNodeAnySetter
+    {
+        String a;
+        JsonNode anySetterNode;
+
+        @JsonCreator
+        public PojoWithNodeAnySetter(@JsonProperty("a") String a,
+            @JsonAnySetter JsonNode leftovers
+        ) {
+            this.a = a;
+            anySetterNode = leftovers;
+        }
+    }
+
+    static class MultipleAny562
+    {
+        @JsonCreator
+        public MultipleAny562(@JsonProperty("a") String a,
+            @JsonAnySetter Map<String, Object> leftovers,
+            @JsonAnySetter Map<String, Object> leftovers2) {
+            throw new Error("Should never get here!");
+        }
+    }
+
+    static class PojoWithDisabled
+    {
+        String a;
+        Map<String,Object> stuff;
+
+        @JsonCreator
+        public PojoWithDisabled(@JsonProperty("a") String a,
+            @JsonAnySetter(enabled = false) Map<String, Object> leftovers
+        ) {
+            this.a = a;
+            stuff = leftovers;
+        }
+    }
+
+    private final ObjectMapper MAPPER = newJsonMapper();
+
+    @Test
+    public void mapAnySetterViaCreator562() throws Exception
+    {
+        Map<String, Object> expected = new HashMap<>();
+        expected.put("b", Integer.valueOf(42));
+        expected.put("c", Integer.valueOf(111));
+
+        POJO562 pojo = MAPPER.readValue(a2q(
+                "{'a':'value', 'b':42, 'c': 111}"
+                ),
+                POJO562.class);
+
+        assertEquals("value", pojo.a);
+        assertEquals(expected, pojo.stuff);
+
+        // Should also initialize any-setter-Map even if no contents
+        pojo = MAPPER.readValue(a2q("{'a':'value2'}"), POJO562.class);
+        assertEquals("value2", pojo.a);
+        assertEquals(new HashMap<>(), pojo.stuff);
+    }
+
+    @Test
+    public void testNodeAnySetterViaCreator562() throws Exception
+    {
+        PojoWithNodeAnySetter pojo = MAPPER.readValue(
+                a2q("{'a':'value', 'b':42, 'c': 111}"),
+                PojoWithNodeAnySetter.class);
+
+        assertEquals("value", pojo.a);
+        assertEquals(a2q("{'c':111,'b':42}"), pojo.anySetterNode + "");
+    }
+
+    @Test
+    public void testAnyMapWithNullCreatorProp() throws Exception
+    {
+        ObjectMapper failOnNullMapper = jsonMapperBuilder()
+            .enable(DeserializationFeature.FAIL_ON_NULL_CREATOR_PROPERTIES).build();
+
+        // 08-Jun-2024, tatu: Should be fine as we get empty Map for "no any setters"
```

Comment:
With change I made wrt ordering (handle buffered any props before null checking), an empty Map will be created for Any property for this case, no null should remain.